### PR TITLE
add edit module settings endpoint

### PR DESCRIPTION
**Pull request by grinakate**
http://outsourcing.nat.tepkom.ru:20080/redmine/issues/968

### CODE REVIEW

File: src/EmailService.Business/Commands/ModuleSetting/EditModuleSettingCommand.cs [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+
+      if (!_validator.ValidateCustom(patch, out List<string> errors))
+      {
+        return _responseCreator.CreateFailureResponse<bool>(HttpStatusCode.BadRequest, errors);
+      }
+
+      OperationResultResponse<bool> response = new();
+
+      response.Body = await _repository.EditAsync(moduleSettingId, _mapper.Map(patch));
+      response.Status = OperationResultStatusType.FullSuccess;
```
Comment (MariyaSin) [inline src/EmailService.Business/Commands/ModuleSetting/EditModuleSettingCommand.cs]: successful status is set by default

File: src/EmailService.Validation/Validators/ModuleSetting/EditModuleSettingRequestValidator.cs [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+      #endregion
+
+      #region Host
+
+      AddFailureForPropertyIf(
+        nameof(EditModuleSettingRequest.Host),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => string.IsNullOrWhiteSpace(x.value?.ToString().Trim()), "Host must not be empty." },
```
Comment (MariyaSin) [inline src/EmailService.Validation/Validators/ModuleSetting/EditModuleSettingRequestValidator.cs]: null ref if null

File: src/EmailService.Validation/Validators/ModuleSetting/EditModuleSettingRequestValidator.cs [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+      AddFailureForPropertyIf(
+        nameof(EditModuleSettingRequest.Password),
+        x => x == OperationType.Replace,
+        new()
+        {
+          { x => !string.IsNullOrWhiteSpace(x.value?.ToString()), "Password must not be empty." },
+          { x => x.value.ToString().Length >= 8, "Password is too short." },
+          { x => x.value.ToString().Length <= 14, "Password is too long." },
+          { x => PasswordRegex.IsMatch(x.value.ToString()), "The password must contain at least one special character." },
+          { x => !x.value.ToString().Contains(' '), "Password must not contain space." }
```
Comment (MariyaSin) [inline src/EmailService.Validation/Validators/ModuleSetting/EditModuleSettingRequestValidator.cs]: check only empty value

File: src/EmailService.Validation/Validators/ModuleSetting/EditModuleSettingRequestValidator.cs [hunk 4]
```diff
@@ -0,0 +1,121 @@
...
+using LT.DigitalOffice.EmailService.Models.Dto.Requests.ModuleSetting;
+using LT.DigitalOffice.EmailService.Validation.Validators.ModuleSetting.Interfaces;
+using LT.DigitalOffice.Kernel.Validators;
+using Microsoft.AspNetCore.JsonPatch.Operations;
+
+namespace LT.DigitalOffice.EmailService.Validation.Validators.ModuleSetting
+{
+  public class EditModuleSettingRequestValidator : BaseEditRequestValidator<EditModuleSettingRequest>, IEditModuleSettingRequestValidator
+  {
+    private static Regex PasswordRegex = new(@"(?=.*[.,:;?!*+%\-<>@[\]{}/\\_{}$#])");
```
Comment (MariyaSin) [inline src/EmailService.Validation/Validators/ModuleSetting/EditModuleSettingRequestValidator.cs]: you do not use it, remove

File: src/EmailService.Mappers/Patch/PatchDbModuleSettingMapper.cs [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+      if (request is null)
+      {
+        return null;
+      }
+
+      JsonPatchDocument<DbModuleSetting> dbPatch = new();
+
+      foreach (var item in request.Operations)
+      {
+        dbPatch.Operations.Add(new Operation<DbModuleSetting>(item.op, item.path, item.from, item.value));
```
Comment (MariyaSin) [inline src/EmailService.Mappers/Patch/PatchDbModuleSettingMapper.cs]: it will be better if you add trim() to item.value

**Review by MariyaSin — APPROVED** [review]

**Review by ekaterinaryzhenkova — APPROVED** [review]

File: src/EmailService.Validation/Validators/ModuleSetting/EditModuleSettingRequestValidator.cs [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        {
+          { x => !string.IsNullOrWhiteSpace(x.value?.ToString()), "Password must not be empty." },
+        });
+
+      #endregion
+    }
+
+    public EditModuleSettingRequestValidator()
+    {
+      RuleForEach(x => x.Operations).Custom(HandleInternalPropertyValidation);
```
Comment (ndoriomedov) [inline src/EmailService.Validation/Validators/ModuleSetting/EditModuleSettingRequestValidator.cs]: move .Custom(...) to next line

**Review by asyagraf — APPROVED** [review]

**Review by ndoriomedov — APPROVED** [review]

**Pull request merged**